src/main/java/com/hazelcast/kubernetes/KubernetesClient.java

```diff
@@ -0,0 +1,290 @@
+    }
+
+    /**
+     * Retrieves POD addresses for all services in the given {@code namespace}.
+     *
+     * @param namespace namespace name
+     * @return all POD addresses from the given {@code namespace}
+     * @see <a href="https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#list-143">Kubernetes Endpoint API</a>
+     */
+    Endpoints endpoints(String namespace) {
+        String urlString = String.format("%s/api/v1/namespaces/%s/endpoints", kubernetesMaster, namespace);
+        return parseEndpointsList(callGet(urlString));
+
+    }
+
+    /**
+     * Retrieves POD addresses for all services in the given {@code namespace} filtered by {@code serviceLabel}
+     * and {@code serviceLabelValue}.
+     *
+     * @param namespace         namespace name
+     * @param serviceLabel      label used to filter responses
+     * @param serviceLabelValue label value used to filter responses
+     * @return all POD addresses from the given {@code namespace} filtered by the label
+     * @see <a href="https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#list-143">Kubernetes Endpoint API</a>
+     */
+    Endpoints endpointsByLabel(String namespace, String serviceLabel, String serviceLabelValue) {
+        String param = String.format("labelSelector=%s=%s", serviceLabel, serviceLabelValue);
+        String urlString = String.format("%s/api/v1/namespaces/%s/endpoints?%s", kubernetesMaster, namespace, param);
+        return parseEndpointsList(callGet(urlString));
+    }
+
+    /**
+     * Retrieves POD addresses from the given {@code namespace} and the given {@code endpointName}.
+     *
+     * @param namespace    namespace name
+     * @param endpointName endpoint name
+     * @return all POD addresses from the given {@code namespace} and the given {@code endpointName}
+     * @see <a href="https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#list-143">Kubernetes Endpoint API</a>
+     */
+    Endpoints endpointsByName(String namespace, String endpointName) {
+        String urlString = String.format("%s/api/v1/namespaces/%s/endpoints/%s", kubernetesMaster, namespace, endpointName);
+        JsonObject json = callGet(urlString);
+        return parseEndpoint(json);
+    }
+
+    private JsonObject callGet(String urlString) {
+        HttpURLConnection connection = null;
+        try {
+            URL url = new URL(urlString);
+            connection = (HttpURLConnection) url.openConnection();
+            if (connection instanceof HttpsURLConnection) {
+                ((HttpsURLConnection) connection).setSSLSocketFactory(buildSslSocketFactory());
+            }
+            connection.setRequestMethod("GET");
+            connection.setRequestProperty("Authorization", String.format("Bearer %s", apiToken));
+
+            if (connection.getResponseCode() != HTTP_OK) {
+                throw new KubernetesClientException(String.format("Failure executing: GET at: %s. Message: %s,", urlString,
```

Comment:
I suppose we reasonably expect exactly 200 for our GETs (not other 2xx codes or even 3xx redirects etc.)

Reply:
Yeah, according to the Kubernetes API Spec [1], `200` is the only valid response.

[1] https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#read-142

Reply:
super!